### PR TITLE
Publish should also support default process

### LINKED ISSUE #599: Default process type is not set when running against a registry
### Summary
Currently, the default process type is only set if publish is false: https://github.com/buildpacks/pack/blob/82549aa7c8980ed7a42b0c8947c0a37d6d1edade/internal/build/phases.go#L191 (note the return on line 185)

We should set the default process type for both the registry and the daemon case.

---

##### Steps

Making some assumptions here...
1. `pack build <some-image> --default-process <some-process-type> --publish`
2. `pack inspect-image` - observe that the default process is not set

##### Current behavior
It is not set.

##### Expected behavior
It should be set.

### PR DESCRIPTION
**Pull request by zmackie**
Fixes #599

### CODE REVIEW

File: internal/build/phases.go [hunk 1]
```diff
@@ -183,31 +187,22 @@ func (l *Lifecycle) newExport(repoName, runImage string, publish bool, launchCac
 		)
 
 		return phaseFactory.New(configProvider), nil
-	}
+	} else {
```
Comment (natalieparellano) [inline internal/build/phases.go:190]: I can see why you added this... this `else` would've likely alerted us all to the initial problem. But now the linter is unhappy :/

**Comment by natalieparellano** [comment]
Besides the failing linter these changes look good to me. Awesome that this is now tested :)

**Comment by jromero** [comment]
@zmackie any follow up on this?

**Comment by zmackie** [comment]
@jromero Should be ready to go. lint fixed.

**Comment by jromero** [comment]
@zmackie seems to still be failing :/

You can try running `make prepare-for-pr` locally.

**Comment by zmackie** [comment]
> @zmackie seems to still be failing :/
> 
> You can try running `make prepare-for-pr` locally.

Should be GTG according to `make prepare-for-pr` (although I think that `make` command is not working properly see:
```
-----------------
NOTICE: There are some files that have not been committed.
-----------------

On branch fix/default-proc-publish
Your branch is up to date with 'origin/fix/default-proc-publish'.

nothing to commit, working tree clean

-----------------
NOTICE: There are some files that have not been committed.
-----------------

```

**Pull request merged**